lib/velum/kubernetes.rb

```diff
@@ -1,9 +1,36 @@
 # frozen_string_literal: true
+
+require "velum/salt"
+
 module Velum
   # Kubernetes deals with the Kubernetes integration of this application.
   class Kubernetes
     attr_reader :client
 
+    KubeConfig = Struct.new :host, :ca_crt, :client_crt, :client_key
+
+    # Returns the Kubernetes apiserver configuration. It returns a KubeConfig struct.
+    def self.kubeconfig
+      _, ca = Velum::Salt.call action:  "mine.get",
+                               targets: "ca",
+                               arg:     ["ca", "x509.get_pem_entries"]
+      _, apiserver_crt = Velum::Salt.call action:      "cmd.run",
+                                          targets:     "roles:kube-master",
+                                          target_type: "grain",
+                                          arg:         ["cat /etc/pki/minion.crt"]
+      _, apiserver_key = Velum::Salt.call action:      "cmd.run",
+                                          targets:     "roles:kube-master",
+                                          target_type: "grain",
+                                          arg:         ["cat /etc/pki/minion.key"]
+      host = Socket.gethostname
```

Comment:
Here we use the hostname of the contoller node as the kubernetes API host. Below, in the initialize method, we use "VELUM_KUBERNETES_HOST" for the same thing, or am I confused?